pgcli/pgexecute.py

```diff
@@ -212,7 +212,11 @@ def connect(
         new_params.update(kwargs)
 
         if new_params["dsn"]:
-            new_params = {"dsn": new_params["dsn"], "password": new_params["password"]}
+            # Preserve hostaddr when using DSN (needed for SSH tunnels with .pgpass)
+            preserved_params = {"dsn": new_params["dsn"], "password": new_params["password"]}
```

Comment:
So here, you're filtering `new_params` down to just 3 keys, `dsn`, `password` and `hostaddr`, is that right? Why do you need another intermediary `preserved_params`?